designs/cloudWatch/cloudWatchIntegration.md

```diff
@@ -0,0 +1,77 @@
+
+*Browsing Log Groups in AWS Explorer:*<br />
+![BROWSE_LOG_GROUP]
+
+#### Alternatives Considered
+Instead of building a new CloudWatch service node in the AWS Explorer, we instead build a CloudWatch Log Group browser tab in the CloudWatch tool window. This new tab is also be linked to the active AWS connection settings. This differs from the other tabs which may lead to a confusing user experience and seen as a con.
+
+*Possible alternative for browsing Log Groups:*<br />
+![ALT_BROWSE_LOG_GROUP]
+
+### View Log Streams
+Upon selecting a Log Group, a new tab in the CloudWatch tool window is shown to the user. This new tab lists all the streams for the requested group to allow the user to find the exact stream they are looking for.
+
+Log Streams can also be sorted by name or last event time in order to help facilitate finding the Log Stream of interest.
+
+*View all Log Streams in the selected Log Group:*<br />
+![BROWSE_LOG_STREAMS]
+
+#### Navigate to a Log Stream
+Double clicking the Log group in the AWS Explorer opens the Log Stream selection. 
+
+For additional service integrations, such as a Lambda function, where we know what Log Group is associated with the resource a new context menu action adds the ability to bypass having to search for the Log Group and instead open the Log Stream selection directly.
+
+### View A Log Stream
+Each Log Stream the user selects opens in its own tab to allow for browsing multiple streams at the time. The Log Stream tab is scoped to the region and credentials used to open it so that the user can browse across regions and accounts at the same time.
+
+*View of a Log Stream:*<br />
+![VIEW_LOG_STREAM]
+
+#### Treat as...
+The Treat As feature acts as a Log Event post-processor to convert the Log Event into a more readable format. This feature provides a pre-defined list of formats such as JSON, Plain Text, CSV, or [CLF](https://httpd.apache.org/docs/1.3/logs.html#common). The formats process each Log Event and convert them into a more tabular format where each column maps to an extract format. The filter feature filters on these extracted values.
+
+*Treat As -> JSON:*<br />
+![VIEW_LOG_STREAM_JSON]
+
+#### View events around...
+In order to facilitate easier triaging, users should be able to search for events such as "Exception" or "Error". This filters the currant Log Stream's events to events that contain the specified string. The user can then right click a Log Entry and select `View events around...`. This context menu action provides a list of pre-canned durations (i.e. `30 seconds`, `1 minute`, `5 minutes`, `custom`) that opens up a new Log Stream tab with that duration pre-selected. This allows the user to see events that led up to and resulted from the error they were searching for without losing their triage progress.
+
+#### Tail Stream
+TODO
```

Comment:
Is this TODO as in "p0 and TBD" or are you saying this will be done post launch?

I see a (the most?) common dev cycle being:

1. Do something locally
2. Check logs

I would rank this as ultra important.

Reply:
It is p0, just I am getting stuck on part of the UX, if the customer is in the middle of a stream, and we have not loaded the data to the end, but is the behavior when they enable tail? Do we:

1. Drop all data and start clean in tail mode as if they "connected" to a live stream?
2. Do we attempt to load all data till now (this seems expensive)
3. Is tail a special mode that acts 100% different than viewing historical data?

I can't decide since everything has pros/cons

Reply:
\4. show a "page break" sigil and start tailing data beneath it.

```
line 1
line 2
[...]  // tail enabled
line 5234233
line 5234234
```